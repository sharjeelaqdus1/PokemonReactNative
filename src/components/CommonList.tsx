// src/components/CommonList.tsx

import React from 'react';
import {FlatList, ListRenderItem, StyleProp, ViewStyle} from 'react-native';

type CommonListProps<T> = {
  data: T[];
  renderItem: ListRenderItem<T>;
  keyExtractor: (item: T, index: number) => string;
  style?: StyleProp<ViewStyle>;
  ListHeaderComponent?: React.ComponentType<any> | React.ReactElement | null;
  ListFooterComponent?: React.ComponentType<any> | React.ReactElement | null;
  ItemSeparatorComponent?: React.ComponentType<any> | null;
  onEndReached?: (() => void) | null;
  onEndReachedThreshold?: number | null;
  testID: string;
};

function CommonList<T>({
  data,
  renderItem,
  keyExtractor,
  style,
  ListHeaderComponent,
  ListFooterComponent,
  ItemSeparatorComponent,
  onEndReached,
  onEndReachedThreshold,
}: CommonListProps<T>) {
  return (
    <FlatList
      data={data}
      renderItem={renderItem}
      keyExtractor={keyExtractor}
      style={style}
      ListHeaderComponent={ListHeaderComponent}
      ListFooterComponent={ListFooterComponent}
      ItemSeparatorComponent={ItemSeparatorComponent}
      onEndReached={onEndReached}
      onEndReachedThreshold={onEndReachedThreshold}
    />
  );
}

export default CommonList;
