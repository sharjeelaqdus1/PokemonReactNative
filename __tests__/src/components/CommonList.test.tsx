import React from 'react';
import {render} from '@testing-library/react-native';
import CommonList from '../../../src/components/CommonList';
import {Text} from 'react-native';

// Mock data for testing
const mockData = [
  {id: '1', name: 'Item 1'},
  {id: '2', name: 'Item 2'},
  {id: '3', name: 'Item 3'},
];

// Mock renderItem function
const renderItem = ({item}: {item: {id: string; name: string}}) => (
  <Text>{item.name}</Text>
);

test('renders correctly', () => {
  const {toJSON} = render(
    <CommonList
      data={mockData}
      renderItem={renderItem}
      keyExtractor={item => item.id}
      testID="commonList"
    />,
  );

  expect(toJSON()).toMatchSnapshot();
});
