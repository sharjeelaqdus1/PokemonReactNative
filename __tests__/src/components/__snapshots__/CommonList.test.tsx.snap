// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  data={
    [
      {
        "id": "1",
        "name": "Item 1",
      },
      {
        "id": "2",
        "name": "Item 2",
      },
      {
        "id": "3",
        "name": "Item 3",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={0.0001}
  stickyHeaderIndices={[]}
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text>
        Item 1
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text>
        Item 2
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text>
        Item 3
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
